core/run_ctest/run_single_ctest.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python3
+import sys, time
+
+from program_input import p_input
+from main import test_conf_file
+from parse_input import *
+
+run_mode = p_input["run_mode"]
+project = p_input["project"]
+mapping = parse_mapping(p_input["mapping_path"])
+from run_test import run_test_batch
+
+def main(argv):
+    print(">>>>[ctest_core] running project {}".format(project))
+    s = time.time()
+    ctestname = argv[1]
+    if run_mode == "run_ctest":
+        print("Please use the run_single_ctest mode")
+        return 
+    elif run_mode == "run_single_ctest":
+        param_value_dict = {}
+        for i in range(2, len(argv)):
+            param, value = argv[i].split('=')
+            param_value_dict[param] = value
+        test_input = extract_conf_diff_from_pair(param_value_dict)
+        test_conf_file(test_input, ctestname)
+        print(">>>>[ctest_core] total time: {} seconds".format(time.time() - s))
+
+def test_conf_file(test_input, ctestname):
+    params = test_input.keys()
+    associated_test_map = dict([[i, list((ctestname, ))] for i in params if ctestname in mapping[i]])
```

Comment:
Minor: RHS could be simplified using dictionary comprehension. Took me a little while to understand what's going on

`{p: [ctestname] for p in params if ctestname in mapping[p]}`